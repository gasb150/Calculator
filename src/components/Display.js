import React from 'react';
import PropTypes from 'prop-types';

const Display = (props) => {
  const { result } = props;
  return (
    <h1 className="ui orange header display-value">
      {result || '0'}
    </h1>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
