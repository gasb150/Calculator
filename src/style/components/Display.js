import React from 'react'

export default class Display extends React.Component {

  state = { showResult: '0' }

  render() {
  return (
    <h1>{this.state.showResult}</h1>
    )
}
}