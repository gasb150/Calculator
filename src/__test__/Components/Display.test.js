import { create } from 'react-test-renderer';

import Display from '../../components/Display';

it('Should render correctly', () => {
  const elem = create(<Display result="some value" />);
  expect(elem).toMatchSnapshot();
});
