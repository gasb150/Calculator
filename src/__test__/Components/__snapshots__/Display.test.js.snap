// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<h1
  className="ui orange header display-value"
>
  some value
</h1>
`;
